main/createUiDefinition.json

```diff
@@ -0,0 +1,153 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/0.1.2-preview/CreateUIDefinition.MultiVm.json#",
+  "handler": "Microsoft.Azure.CreateUIDef",
+  "version": "0.1.2-preview",
+  "parameters": {
+    "basics": [
+      {
+        "name": "vmName",
+        "type": "Microsoft.Common.TextBox",
+        "label": "Bloxone VM name",
+        "defaultValue": "Bloxone",
+        "toolTip": "The name of the new Bloxone vm.",
+        "multiLine": false,
+        "constraints": {
+          "required": true,
+          "regex": "^(([0-9a-zA-Z][0-9a-zA-Z-.]{0,63}[0-9a-zA-Z])|[0-9a-zA-Z]).{1,64}$",
+          "validationMessage": "The length must be between 1 and 64 characters. The first and the last characters must be letters or numbers. The remaining characters must be letters, numbers, periods or dashes."
+        },
+        "visible": true
+      },
+      {
+        "name": "adminPassword",
+        "toolTip": "Password for the user",
+        "type": "Microsoft.Common.PasswordBox",
+        "label": {
+          "password": "Password",
+          "confirmPassword": "Confirm password"
+        },
+        "constraints": {
+          "required": true,
+          "regex": "^(?:(?=.*[a-z])(?:(?=.*[A-Z])(?=.*[\\d\\W])|(?=.*\\W)(?=.*\\d))|(?=.*\\W)(?=.*[A-Z])(?=.*\\d)).{12,64}$",
+          "validationMessage": "The password must be between 12 and 64 characters long, and contain characters from at least 3 of the following groups: uppercase characters, lowercase characters, numbers, and special characters."
+        },
+        "visible": true
+      },
+      {
+        "name": "customData",
+        "type": "Microsoft.Common.TextBox",
+        "label": "Custom data",
+        "toolTip": "Pass custom data configuration.",
```

Comment:
what is the difference between custom-data vs cloud-init `userdata` but we are passing it right?

Reply:
Azure supports two formats: `User data` and `Custom data`. As written in the documentation:
- [User data](https://docs.microsoft.com/en-us/azure/virtual-machines/user-data) is a new version of [custom data](https://docs.microsoft.com/en-us/azure/virtual-machines/custom-data)

And yes we are passing our userdata to Azure customData and it works fine. Our userdata does not work with new version `User data` of Azure.